## Importing modules and data and splitting

In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv(r"D:\Sem__Project\Sem8\DataPrepared\data0Prepared.csv")
data = dataset.copy()

FileNotFoundError: [Errno 2] No such file or directory: 'D:\\Sem__Project\\Sem8\\DataPrepared\\data0Prepared.csv'

In [ ]:
label = data.iloc[:, -1]
feats = data.iloc[:, :-1]
feats.columns

Index(['latitude', 'longitude', 'twi', 'tpi', 'slope', 'rainfall', 'hillshade',
       'elevation', 'drainageDensity', 'profileCurvature', 'planCurvature',
       'curvature', 'aspect', 'LULC1', 'LULC2', 'LULC5', 'LULC7', 'LULC8',
       'LULC9', 'LULC10', 'LULC11', 'Soil1', 'Soil2', 'Soil3', 'Soil4',
       'Soil5'],
      dtype='object')

In [ ]:
from sklearn.model_selection import train_test_split
xTrain, xTest, yTrain, yTest = train_test_split(feats, label, test_size=0.30, random_state=42, stratify=label)
xTrain.shape, xTest.shape


((156, 26), (68, 26))

In [ ]:
def showMetrics(yPred):
    from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
    from sklearn.metrics import roc_curve, auc
    print('Confusion Matrix:\n', confusion_matrix(yTest, yPred))
    print('Accuracy Score:', accuracy_score(yTest, yPred))
    print('Precision Score:', precision_score(yTest, yPred))
    print('Recall Score:', recall_score(yTest, yPred))
    print('F1 Score:', f1_score(yTest, yPred))
    FPR, TPR, thres = roc_curve(yTest, yPred)
    print('ROC AUC score:', auc(FPR, TPR))

# Random Forest Classification

In [ ]:
from sklearn.ensemble import RandomForestClassifier
rfc = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
rfc.fit(xTrain, yTrain)
yPred = rfc.predict(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[38  4]
 [ 8 18]]
Accuracy Score: 0.8235294117647058
Precision Score: 0.8181818181818182
Recall Score: 0.6923076923076923
F1 Score: 0.75
ROC AUC score: 0.7985347985347985


# Logistic Regression

In [ ]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression(random_state=0)
lr.fit(xTrain, yTrain)
yPred = lr.predict(xTest)
yPred_lr = lr.decision_function(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[37  5]
 [ 4 22]]
Accuracy Score: 0.8676470588235294
Precision Score: 0.8148148148148148
Recall Score: 0.8461538461538461
F1 Score: 0.8301886792452831
ROC AUC score: 0.8635531135531136


# K-Nearest Neighbors (KNN)

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=5, metric='minkowski', p=2)
knn.fit(xTrain, yTrain)
yPred = knn.predict(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[34  8]
 [10 16]]
Accuracy Score: 0.7352941176470589
Precision Score: 0.6666666666666666
Recall Score: 0.6153846153846154
F1 Score: 0.64
ROC AUC score: 0.7124542124542125


# Support Vector Machine (SVM)

In [ ]:
from sklearn.svm import SVC
svmLin = SVC(kernel='linear', random_state=0)
svmLin.fit(xTrain, yTrain)
yPred = svmLin.predict(xTest)
yPred_svmLin = svmLin.decision_function(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[40  2]
 [ 7 19]]
Accuracy Score: 0.8676470588235294
Precision Score: 0.9047619047619048
Recall Score: 0.7307692307692307
F1 Score: 0.8085106382978723
ROC AUC score: 0.8415750915750916


# Kernal SVM

In [ ]:
from sklearn.svm import SVC
svmRBF = SVC(kernel='rbf', random_state=0)
svmRBF.fit(xTrain, yTrain)
yPred = svmRBF.predict(xTest)
yPred_svmRBF = svmRBF.decision_function(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[39  3]
 [10 16]]
Accuracy Score: 0.8088235294117647
Precision Score: 0.8421052631578947
Recall Score: 0.6153846153846154
F1 Score: 0.7111111111111111
ROC AUC score: 0.771978021978022


# Naive Bayes

In [ ]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(xTrain, yTrain)
yPred = nb.predict(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[29 13]
 [ 3 23]]
Accuracy Score: 0.7647058823529411
Precision Score: 0.6388888888888888
Recall Score: 0.8846153846153846
F1 Score: 0.7419354838709677
ROC AUC score: 0.7875457875457876


# Decision Tree Classification

In [ ]:
from sklearn.tree import DecisionTreeClassifier
dtc = DecisionTreeClassifier(criterion='entropy', random_state=0)
dtc.fit(xTrain, yTrain)
yPred = dtc.predict(xTest)
showMetrics(yPred)

Confusion Matrix:
 [[37  5]
 [ 7 19]]
Accuracy Score: 0.8235294117647058
Precision Score: 0.7916666666666666
Recall Score: 0.7307692307692307
F1 Score: 0.76
ROC AUC score: 0.8058608058608059


# Best Classification by Logistic Regression
Confusion Matrix:<br>
 [[37  5]<br>
 [ 4 22]]<br>
Accuracy Score: 0.8676470588235294<br>
Precision Score: 0.8148148148148148<br>
Recall Score: 0.8461538461538461<br>
F1 Score: 0.8301886792452831<br>
ROC AUC score: 0.8635531135531136<br>